packages/server/src/queries/complex/portfolio/allocation.ts

```diff
@@ -0,0 +1,184 @@
+import { CoinPretty } from "@keplr-wallet/unit";
+import { AssetList } from "@osmosis-labs/types";
+import cachified, { CacheEntry } from "cachified";
+import { LRUCache } from "lru-cache";
+
+import { queryAllocation } from "../../../queries/data-services";
+import { Categories } from "../../../queries/data-services";
+import { AccountCoinsResult } from "../../../queries/data-services";
+import { DEFAULT_LRU_OPTIONS } from "../../../utils/cache";
+import { getAsset } from "../assets";
+
+const allocationCache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+
+interface FormattedAllocation {
+  key: string;
+  percentage: number;
+  amount: number;
+  asset?: CoinPretty;
+}
+
+export interface GetAllocationResponse {
+  all: FormattedAllocation[];
+  assets: FormattedAllocation[];
+  available: FormattedAllocation[];
+}
+
+async function getAll(categories: Categories) {
+  const userBalancesCap = +categories["user-balances"].capitalization;
+  const stakedCap = +categories["staked"].capitalization;
+  const unstakingCap = +categories["unstaking"].capitalization;
+  const unclaimedRewardsCap = +categories["unclaimed-rewards"].capitalization;
+  const pooledCap = +categories["pooled"].capitalization;
```

Comment:
I think it would be a good idea to Dec types to do calculations. It uses the `BigInt` type under the hood. It's a 1-1 mapping of the Dec type used by SQS.

I looked at the query response, and the capitalization responses contain 18 decimals, which may cause precision issues if used directly with normal JS number type.

Also, these Dec types are directly passable to CoinPretty, which would be a better more useful type to return the coin amounts as.